src/api/inventory.js

```diff
@@ -63,8 +63,40 @@ const mapData = ({ results, ...data }) => ({
     }))
 });
 
-export function getEntities(items, { base = INVENTORY_API_BASE }) {
-    return fetch(`${base}${items.length !== 0 ? '/' + items : ''}`).then(r => {
+function buildQuery({ per_page, page, filters }) {
+    const allowedFilters = [ 'display_name' ];
+    let query = [];
+    const makeValue = (item, keyValue, keyFilter) => (
+        allowedFilters.find(allowed => allowed === item[keyValue]) && `${item[keyValue]}=${item[keyFilter]}`
+    );
+    if (per_page || page) {
+        const params = { per_page, page };
+        query = [
+            ...query,
+            ...Object.keys(params).reduce(
+                (acc, curr) => [ ...acc, `${curr}=${params[curr]}` ], []
+            )
+        ];
+    }
+
+    if (filters) {
+        query = [
+            ...query,
+            ...Object.values(filters).reduce((acc, curr) => [
+                ...acc,
+                makeValue(curr, 'value', 'filter'),
+                makeValue(curr, 'group', 'value')
+            ], [])
+        ].filter(Boolean);
+    }
+
+    return query ? `?${query.join('&')}` : '';
+}
+
+export function getEntities(items, { base = INVENTORY_API_BASE, ...rest }) {
+    let query = buildQuery(rest);
+
+    return fetch(`${base}${items.length !== 0 ? '/' + items : ''}${query}`).then(r => {
```

Comment:
I believe this is building the wrong query for 'tree-like' Inventory tables. Notice in such tables, you would have an items array like the following: https://github.com/RedHatInsights/insights-frontend-components/pull/168/files#diff-99c7484e5cb238b566e93f5654bae0e7R290

All the collapsible rows don't have any ID, therefore, this is constructing a query with 'some-id,' (finishing with a comma), which returns a 500 from the inventory.

A simple `items.filter(n => n)` here fixes the issue.

Reply:
Well it does fix the issue.. for the inventory table. For inventory detail, `items.filter` wouldn't work because `items` here is just an ID. I suppose it would have to be dealt with differently. Maybe making an array `[items]` then flattening it, so it works for both.

Reply:
```javascript
function flatten(a) {
    return Array.isArray(a) ? [].concat(...a.map(flatten)) : a;
}

export function getEntities(items, { base = INVENTORY_API_BASE, ...rest }) {
    let query = buildQuery(rest);

    return fetch(`${base}${items.length !== 0 ? '/' + flatten([ items ].filter(n => n)) : ''}${query}`)
 ```

this worked

Reply:
This function expects items to be either one ID or array of IDs, there is function in [InventoryList.js#L97](https://github.com/RedHatInsights/insights-frontend-components/blob/f4c5b0b79195445652891d24772ff4bfd57897de/src/SmartComponents/Inventory/InventoryList.js#L97) which does something similiar to what you are proposing. So no need to do this twice since I wanted to keep this function as simple as possible.

Reply:
I don't think this works with one ID, really. It does work if it contains an array of IDs with one ID. Using just an ID, the "," at the end of the URL breaks the inventory API call. 

 GET https://prod.foo.redhat.com:1337/r/insights/platform/inventory/api/v1/hosts/ed9b9afd-4490-4ba8-b1a5-af63fd34c124, - 500 (Internal Server Error) (note the dangling comma)

![screenshot from 2019-01-02 11-40-36](https://user-images.githubusercontent.com/598891/50588641-4903c800-0e83-11e9-8527-95f65be921e6.png)

If you don't want to change this function, perhaps whatever calls `getEntities` should be aware that 'items' can look like [id1], or [id1, id2, id3], but not simply 'id1' as the inventory does not respond to that

Reply:
Ooh, I see your point now. Yeah items has to be array of strings, not just plain string, I am gonna add error throw if incorrect data type is sent to inventory to notify consumers how it should look like. Good catch however!